### ISSUE TITLE
~/.config/qutebrowser/blocked-hosts misinterpretation

### ISSUE THREAD

**Issue by Zeadar** [qutebrowser/qutebrowser#4473]
**Version info (see `:version`)**:
qutebrowser v1.5.2
Git commit: 
Backend: QtWebEngine (Chromium 65.0.3325.230)

CPython: 3.7.1
Qt: 5.11.2
PyQt: 5.11.3

sip: 4.19.13
colorama: 0.4.0
pypeg2: 2.15
jinja2: 2.10
pygments: 2.2.0
yaml: 3.13
cssutils: no
attr: 18.2.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: yes
pdf.js: 2.0.943 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.25.3
QtNetwork SSL: OpenSSL 1.1.1  11 Sep 2018

Style: Qt5CTProxyStyle
Platform: Linux-4.4.163-1-MANJARO-x86_64-with-arch-Manjaro-Linux, 64bit
Linux distribution: Manjaro Linux (manjaro)
Frozen: False
Imported from /usr/lib/python3.7/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /home/zeadar/.cache/qutebrowser
config: /home/zeadar/.config/qutebrowser
data: /home/zeadar/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Uptime: 0:00:02

**Description**
qutebrowser fails to block (or redirect) an url that is stated in with the `~/.config/qutebrowser/blocked-hosts` file by the format `127.0.0.1 www.youtube.com`. It should be noted that adding just `www.youtube.com` works as intended.

**How to reproduce**
Add `127.0.0.1 www.youtube.com`
run :adblock-update or restart qutebrowser
Successfully watch cat videos on youtube.

**Comment by gebulmer** [comment]
@The-Compiler This might be a good first task as it's quite short and sweet, so unless I'm mistaken I'll leave it to someone else to actually do

I suspect the issue is entirely within `qutebrowser/browser/adblock.py`, and is due to the config blocked hosts file lines not being parsed but just directly added to the set to check against. This leads to `127.0.0.1 www.youtube.com` being stored instead of what we (probably) want which is just `www.youtube.com`

If I'm right, the newcomer fixing this needs to:
* Start by reading `adblock.py` first
* Take a look at `_parse_line` in that file for the two format cases we've used elsewhere (one host per line and /etc/hosts style)
* Take a look at `_read_hosts_file` in that file and consider what the lines may look like
* Change `_read_hosts_file` to handle that in a similar way to `_parse_line`

Bonus points (imo) for refactoring `_parse_line` to return a list, empty or otherwise, of hosts it finds - remembering to modify `_merge_file`'s use of it, and removing the `try/except ValueError`

Finally: (harder part)
* Read `tests/unit/browser/test_adblock.py` and consider adding or modifying a test for this case

**Comment by prophetofxenu** [comment]
I started working on this. I was able to:
* Change `_read_hosts_file` to be more like `_parse_line`
* Refactor `_parse_line` to return a list of found hosts (and update `_merge_file`) as well as implement `find` instead of a `try/except ValueError`

I was unable to reproduce https://github.com/qutebrowser/qutebrowser/issues/4325. I believe my changes would fix this problem anyway but I was unable to test if so.

I tested this own my own by adding `127.0.0.1 www.youtube.com` to `~/.config/qutebrowser/blocked-hosts`. Before, YouTube would not be blocked by this addition. Afterwards, YouTube is properly blocked.

I did not add any test cases for my changes. I'm not familiar with pytest and I did not feel I could properly write any tests after looking through `test_adblock.py`. However, I did sucessfully run all existing tests for the adblock module.

I apologize for not leaving a comment before I started to work on this. I didn't know about the CONTRIBUTING.asciidoc rules and this is also my first time contributing to an open source project :)
